pkg/decision/experiment_override_service.go

```diff
@@ -62,10 +81,31 @@ func (m *MapExperimentOverridesStore) GetVariation(overrideKey ExperimentOverrid
 }
 
 // SetVariation sets the given variation key as an override for the given user+experiment key
-func (m *MapExperimentOverridesStore) SetVariation(overrideKey ExperimentOverrideKey, variationKey string) {
-	m.mutex.Lock()
-	m.overridesMap[overrideKey] = variationKey
-	m.mutex.Unlock()
+func (m *MapExperimentOverridesStore) SetVariation(overrideKey ExperimentOverrideKey, variationKey string) bool {
+
+	assignVariationKey := func() {
+		m.mutex.Lock()
+		m.overridesMap[overrideKey] = variationKey
+		m.mutex.Unlock()
+	}
+	// Assign directly if no config provided
+	if m.config == nil {
+		assignVariationKey()
+		return true
+	}
+	// Check if experiment and variation exist
+	if experiment, err := m.config.GetExperimentByKey(overrideKey.ExperimentKey); err == nil {
```

Comment:
Hmm I am not sure this is a valid thing to do. The project config is not static. It can change at any time and this reference would be stale. This experiment/variation validation is already being done in `GetDecision` so we don't really have to have this check during the setting of the override

Reply:
One of the tests in fsc sets a valid variation first, then tries setting an invalid variation and expects variation to be not set. The expected result is the first variation set. This test fails with the previous implementation if we dont check for valid variation during setVariation method.

Reply:
@mjc1283 what do you think? This is a bit cumbersome especially if we do check it at the bottom, and granted that it's a QA tool, I don't think we should over-engineer